haystack/document_stores/opensearch.py

```diff
@@ -451,6 +483,17 @@ def query_by_embedding(
             self._convert_es_hit_to_document(hit, adapt_score_for_embedding=True, scale_score=scale_score)
             for hit in result
         ]
+
+        if self.index_type == "hnsw":
+            ef_search = 20 if "ef_search" not in self.knn_parameters else self.knn_parameters["ef_search"]
```

Comment:
This line is duplicated 3 times throughout the module. I recommend to introduce a function for this.